lib/wibox/widget/template.lua

```diff
@@ -134,44 +158,166 @@ function template:update(args)
 end
 
 --- Change the widget template.
--- @tparam table|widget widget_template The new widget to use as a
+--
+-- Note that this will discard the existing widget instance. Thus, any
+-- `set_property` will no longer be honored. `bind_property`, on the other hand,
+-- will still be honored.
```

Comment:
The connections stay intact, but until the signal fires again, the value will stay `nil`. This should be made clear, otherwise users might assume that `bind_property` values are updated immediately on the new template.